### PR TITLE
Logout Button

### LINKED ISSUE #161: add logout button
- [x] Add Button to popovermenu
- [x] Logout function to startpage, delete all settings
- [x] Unit Tests
  - [x] Storage remove im Setting service
  - [x] Logout function in Popover menu with deleting user data
- [x] End2End-Tests

### PR DESCRIPTION
**Pull request by mleu**
Add a logout button. 
Closes #161

### CODE REVIEW

File: e2e/logout.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  beforeEach(() => {
+    browser.get('');
+  });
+
+  afterEach(function () {
+    browser.manage().deleteAllCookies();
+    browser.executeScript('window.sessionStorage.clear();');
+    browser.executeScript('window.localStorage.clear();');
+    browser.executeScript('window.indexedDB.deleteDatabase("imsClientDB")');
```
Comment (sandrozbinden) [inline e2e/logout.e2e-spec.ts]: Code duplication same code exits in settings e2e

File: e2e/logout.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  });
+});
+
+function login() {
+  serverInput.sendKeys('https://sinv-56028.edu.hsr.ch');
+  userInput.sendKeys('admin');
+  passwordInput.sendKeys('admin');
+  loginButton.click();
+  browser.waitForAngular();
+}
```
Comment (sandrozbinden) [inline e2e/logout.e2e-spec.ts]: Code duplication same code exits in settings e2e

File: src/pages/popover/popover.ts [hunk 3]
```diff
@@ -1,18 +1,27 @@
...
+  constructor(public navCtrl: NavController, public navParams: NavParams, public viewCtrl: ViewController, public settingService: SettingService) { }
 
 
   loadSettings() {
     this.navCtrl.push(SettingsPage);
     this.viewCtrl.dismiss();
   }
+
+  logoutUser() {
+    this.settingService.clearLoginData().subscribe();
```
Comment (sandrozbinden) [inline src/pages/popover/popover.ts]: I think this is wrong. It has nothing to do with the persistent settings when we log out a user. What we need to do is clearing the credentials in the AuthService

File: src/providers/setting-service.spec.ts [hunk 4]
```diff
@@ -6,27 +6,40 @@ import { SettingService } from './setting-service';
 
 describe('Provider: SettingService', () => {
 
-    beforeEach(async(() => {
-
-        TestBed.configureTestingModule({
```
Comment (sandrozbinden) [inline src/providers/setting-service.spec.ts]: There might be an issue with line endings. Why is this a diff ?

Comment (mleu) [inline src/providers/setting-service.spec.ts]: Indent was changed from four to two spaces.

**Comment by mleu** [comment]
Oops looks like we where still mentally in the settings branch when building it this way...
New PR #229 uses Auth Service.

**Comment by mleu** [comment]
Obsoleted by #229. We learned a lot 🥇

**Pull request closed without merging**